src/maxdiffusion/models/train.py

```diff
@@ -276,21 +305,30 @@ def compute_loss(unet_params):
     max_logging.log(f"number parameters: {num_model_parameters/10**9:.3f} billion")
 
     my_data_sharding = {'input_ids': data_sharding, 'pixel_values': data_sharding}
-    if (not config.enable_profiler):
-        with jax.transfer_guard("disallow"):
-            p_train_step = jax.jit(
-                partial(train_step, cache_latents_text_encoder_outputs=config.cache_latents_text_encoder_outputs),
-                in_shardings=(unet_state_mesh_shardings, vae_state_mesh_shardings, my_data_sharding, None),
-                out_shardings=(unet_state_mesh_shardings, None, None),
-                donate_argnums=(0,)
-            )
-    else:
+
+    with mesh, nn_partitioning.axis_rules(config.logical_axis_rules):
         p_train_step = jax.jit(
             partial(train_step, cache_latents_text_encoder_outputs=config.cache_latents_text_encoder_outputs),
             in_shardings=(unet_state_mesh_shardings, vae_state_mesh_shardings, my_data_sharding, None),
             out_shardings=(unet_state_mesh_shardings, None, None),
             donate_argnums=(0,)
         )
+        max_logging.log("Precompiling...")
+        s = time.time()
+        dummy_batch = get_shaped_batch(config, pipeline)
```

Comment:
Do we need flag to turn off pre-compilcation in case the training  data format is different from what we defined?

Reply:
I gave it some thought and found it that we should enforce best practices instead of providing users too many options. At the end, the training and inference scripts are reference implementations of a task and it should be expected for a user to modify or create new scripts for their corner cases.